src/components/FeatureStyler/FeatureStyler.js

```diff
@@ -264,8 +291,32 @@ class FeatureStyler extends PureComponent {
     return colors.find(c => rgb === asString(c.fill));
   }
 
+  static getVertexCoord(geom, start = true, index = 0) {
+    const coords = geom.getCoordinates();
+    const len = coords.length - 1;
+    return start ? coords[index] : coords[len - index];
+  }
+
+  static getLineIcon(geom, icon, color, start = true) {
+    const coordA = FeatureStyler.getVertexCoord(geom, start, 1);
+    const coordB = FeatureStyler.getVertexCoord(geom, start);
+    const dx = coordA[0] - coordB[0];
+    const dy = coordA[1] - coordB[1];
+    const rotation = Math.atan2(dy, dx);
+
+    return new Style({
+      geometry: new Point(coordB),
```

Comment:
If you use a geometryFunction  , the arrow should move with the line properly

Reply:
Good idea!
Done.